test_programs/compile_failure/cast_numeric_to_bool/src/main.nr

```diff
@@ -0,0 +1,10 @@
+fn main() {
+    let x = 1;
+    let _ = x as bool;
```

Comment:
We will also want to block this in SSA. The linked issue shows an SSA snippet.

Reply:
I was about to implement this now but I noticed that the `Cast` SSA instruction takes a numeric type. So technically there's no way to cast to bool in SSA, only to `u1`. But now that it's disallowed in the frontend I think all is good.

Reply:
> `Cast` SSA instruction takes a numeric type. So technically there's no way to cast to bool in SSA, only to `u1`. But now that it's disallowed in the frontend I think all is good.

Oh ok so for casting a numeric to a bool we are good then. Can you still trigger the ACIR vs. Brillig difference by directly writing the SSA but with `as u1`? If so this is a separate bug though and we can do it in a follow-up. In general I think we should lean into the SSA always being well formed even if the frontend prevents it (https://github.com/noir-lang/noir/issues/8705).

Reply:
I just checked and with `u1` both programs give 0 as a result.

What's strange is that with `bool` and running in `master` I also get `0` for the brillig case, so I don't know why for the OP it gives 1.